app/util/log.ts

```diff
@@ -1,9 +1,35 @@
+import * as _ from 'lodash';
 import * as winston from 'winston';
-
 import env = require('./env');
 
 const envNameFormat = winston.format((info) => ({ ...info, env_name: env.harmonyClientId }));
 
+/**
+ * Redact sensitive key values from an object.
+ * @param obj - the object to inspect
+ * @param sensitiveKeys - keys for which values will be redacted
+ */
+export function redact(obj: object, sensitiveKeys: RegExp[]): void {
+  Object.keys(obj).forEach(function (key) {
+    if (typeof obj[key] === 'object') {
+      redact(obj[key], sensitiveKeys);
```

Comment:
Good point. I'll fix that.

Reply:
I implemented a fix. The only downside is that if sensitive values exist within the circular reference, those won't get redacted since we will want to abandon ship rather than follow the circular reference round and round. 

However, I don't think that'll be an issue because the sensitive values would still be redacted at the top level of the object, and I think it's likely that the logging libraries we use won't print circular references in full anyway as it would be an infinite amount of text.